src/main/java/fi/fmi/avi/converter/tac/swx/SWXTACParser.java

```diff
@@ -164,7 +166,7 @@ public void setTACLexer(final AviMessageLexer lexer) {
 
         firstLexeme.findNext(LexemeIdentity.REMARKS_START, (match) -> {
             final List<String> remarks = getRemarks(match, hints);
-            if (!remarks.isEmpty()) {
+            if (!remarks.isEmpty() && !remarks.get(0).equalsIgnoreCase("NIL")) {
```

Comment:
The condition must be:

```
!remarks.isEmpty() && (remarks.size() != 1 || !remarks.get(0).equalsIgnoreCase("NIL"))
```

Otherwise `NIL NIL` will be parsed as empty remarks. Though it's unlikely to be a real case.

Reply:
I made the requested changes